app/models/estimated_population.rb

```diff
@@ -26,4 +26,24 @@ def is_population_available_for_all_districts
     end
     is_population_available
   end
+
+  def hypertension_patient_coverage_rate
+    population = region.estimated_population.population.to_f
+    rate = (region.registered_patients.with_hypertension.count.to_f / population) * 100
+    return nil if rate.infinite?
+    return 100.0 if rate > 100.0
+    return rate if rate > 0.0
+  end
+
+  def show_coverage
+    show_coverage = false
+
+    if region.district_region? && region.estimated_population&.hypertension_patient_coverage_rate
+      show_coverage = true
+    elsif region.state_region? && region.estimated_population&.is_population_available_for_all_districts
+      show_coverage = true
```

Comment:
You can refactor to something like this here:

```suggestion
    if region.district_region?
      region.estimated_population&.hypertension_patient_coverage_rate
    elsif region.state_region?
      region.estimated_population&.is_population_available_for_all_districts
    else
      false
    end
```

Reply:
Great, thanks!